compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.td

```diff
@@ -198,7 +199,10 @@ def IREEGPU_MMAAttr : IREEGPU_MmaVectorLayoutAttr<"MMA", "MMAIntrinsicAttr"> {
 
   let skipDefaultBuilders = 1;
   let builders = [
-    AttrBuilder<(ins "MMAIntrinsic":$intrinsic)>
+    AttrBuilder<(ins
+      "MMAIntrinsic":$intrinsic,
+      CArg<"bool", "false">:$isDataTiled
```

Comment:
The `isDataTiled` boolean is functionally the same as adding an `iree_gpu.data_tiled_layout` wrapper class for this early prototype. Right now, the goal is to quickly lay down the base for GPU materialization for just the F32 intrinsic. Once that is working, there will be time to refactor the codegen details. Since this has been blocking progress for a while, and it is only landing to an experimental branch, I think it is fine to do for now. I agree this should not land like this to main, but when things are fleshed out and ready to land, it can be refactored.